src/D2L.CodeStyle.Analyzers/Async/Generator/AsyncToSyncMethodTransformer.cs

```diff
@@ -272,22 +287,35 @@ private VariableDesignationSyntax Transform( VariableDesignationSyntax des )
 				newExpr = memberAccess.Expression;
 				return Transform( newExpr );
 			}
+		} else if( memberAccess is not null && ShouldWrapMemberAccessesInTaskRun( memberAccess ) ) {
+			m_disableTaskRunWarningFlag = true;
+			return SyntaxFactory.ParseExpression( $"Task.Run(() => {invocationExpr}).Result" );
 		}
+
 		return invocationExpr
 			.WithExpression( Transform( invocationExpr.Expression ) )
 			.WithArgumentList( TransformAll( invocationExpr.ArgumentList, Transform ) );
 	}
 
-	bool ShouldRemoveReturnedMemberAccess( MemberAccessExpressionSyntax memberAccessExpr )
-	  => memberAccessExpr.Name.Identifier.ValueText switch {
-		  "FromResult" => true,
-		  "CompletedTask" => true,
-		  _ => false
-	  };
+	bool ShouldRemoveReturnedMemberAccesses( MemberAccessExpressionSyntax memberAccessExpr ) {
+		return ( memberAccessExpr.Expression.ToString(), memberAccessExpr.Name.Identifier.ValueText ) switch
+		{
+			( "Task", "FromResult" ) => true,
+			( "Task", "CompletedTask" ) => true,
+			_ => false
+		};
+	}
+
+	bool ShouldWrapMemberAccessesInTaskRun( MemberAccessExpressionSyntax memberAccessExpr ) {
+		return (memberAccessExpr.Expression.ToString(), memberAccessExpr.Name.Identifier.ValueText) switch {
+			(_, "ReadAsStringAsync" ) => true,
+			_ => false
+		};
+	}

```

Comment:
_Accesses_ in this function name shouldn't be plural. We're using it to ask a question given a _single_ member access expr. If you read it out as such I think it's probably a little more clear

`If I Should Wrap Member Access [memberAccessExpr] In Task Run - Then ...`

```suggestion
	bool ShouldWrapMemberAccessInTaskRun( MemberAccessExpressionSyntax memberAccessExpr ) {
		return (memberAccessExpr.Expression.ToString(), memberAccessExpr.Name.Identifier.ValueText) switch {
			(_, "ReadAsStringAsync" ) => true,
			_ => false
		};
	}
```

Reply:
Thanks!